rest_framework/pagination.py

```diff
@@ -711,7 +711,11 @@ def encode_cursor(self, cursor):
         return replace_query_param(self.base_url, self.cursor_query_param, encoded)
 
     def _get_position_from_instance(self, instance, ordering):
-        attr = getattr(instance, ordering[0].lstrip('-'))
+        field_name = ordering[0].lstrip('-')
+        if isinstance(instance, dict):
+            attr = instance[field_name]
+        else:
+            attr = getattr(instance, field_name)
```

Comment:
Fair enough, good to see that it's a small change. Main Q we need to ask is if we should pull this in for the project, or if we should leave it as a customization for users to make if it is needed.